Bug fix - Lab & Luv color conversion nans in gradient

cceyda opened the pull request:
### Description
Bug fix for https://github.com/kornia/kornia/issues/868
- `torch.pow(x=0)` ops were causing nans. solved by using `x.clamp_min(threshold)` (doesn't effect result values)
- Moved the Linear RGB<->sRGB conversion used in color conversions to rgb module. They are commonly used in luv and lab conversions. Previously the terms sRGB <-> linear rgb were mistakenly referred to in reverse, now fixed [Ref](https://stackoverflow.com/questions/35952564/convert-rgb-to-srgb)

### Status
**Ready**

### Types of changes
<!--- Put an `x` in all the boxes that apply, and remove the not applicable items -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [x] Docstrings/Documentation updated


## PR Checklist
### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [x] Did you discuss the functionality or any breaking changes before ?
- [x] **Pass all tests**: did you test in local ? `make test`
- [x] Unittests: did you add tests for your new functionality ?
- [x] Documentations: did you build documentation ? `make build-docs`
- [x] Implementation: is your code well commented and follow conventions ? `make lint`
- [x] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [x] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

cceyda commented on kornia/color/lab.py:
```diff
@@ -4,6 +4,7 @@
 import torch.nn as nn
 
 from .xyz import rgb_to_xyz, xyz_to_rgb
+from .rgb import rgb_to_linear_rgb, linear_rgb_to_rgb
```
I'm not really liking these function names `rgb_to_linear_rgb, linear_rgb_to_rgb` although `srgb_to_linear_rgb, linear_rgb_to_srgb` would have been more accurate. I stuck with just RGB for consistency. maybe we should mention somewhere in the docs that kornia assumes sRGB as the default RGB format.

edgarriba replied:
and what about simply `srgb_to_rgb` <-> `rgb_to_srgb` ?

cceyda replied:
`srgb_to_rgb` might be more confusing because `linear_rgb != rgb`. 
I'm leaving it as it is because we use `srgb==rgb` anyway

edgarriba commented on kornia/color/lab.py:
```diff
@@ -37,27 +38,20 @@ def rgb_to_lab(image: torch.Tensor) -> torch.Tensor:
 
-    power = torch.pow(xyz_normalized, 1 / 3)
+    threshold = 0.008856
+    power = torch.pow(xyz_normalized.clamp_min(threshold), 1 / 3.)
```
not whether `clamp_min` will be supported in older versions. I guess we could alternatively use `.clamp(min=threshold)`

cceyda replied:
I'm a slave to copy/paste 🤣 replaced all with clip

edgarriba commented on kornia/color/rgb.py:
```diff
@@ -169,6 +169,55 @@ def rgba_to_bgr(image: torch.Tensor) -> torch.Tensor:
 
 
+def rgb_to_linear_rgb(image: torch.Tensor) -> torch.Tensor:
+    r"""Convert an sRGB image to linear RGB. Used in colorspace conversions.
```
ok, given this function definition we should definitely name this function as `srgb_to_rgb`

edgarriba commented on kornia/color/rgb.py:
```diff
@@ -169,6 +169,55 @@ def rgba_to_bgr(image: torch.Tensor) -> torch.Tensor:
+    Example:
+        >>> input = torch.rand(2, 3, 4, 5)
+        >>> output = rgb_to_linear_rgb(input) # 2x3x4x5
+    """
```
could we add basic type and shape check here ?

edgarriba reviewed (commented):
This is great! Minor comments and we need test for the new functions and possibly a specific test that shows that the raised issue is solved. Let me know if you need any help - we are releasing this week and this fix should be definitely in.

cceyda commented:
- added tests for `rgb_to_linear_rgb` &` linear_rgb_to_rgb`. It looked like some stuff for `bgr_to_rgb` were missing in tests so added those I saw was missing. On second thought I decided to leave the function names as is, better be explicit than vague, because I don't want there to be confusion between `sRGB`, `RGB` and `linear RGB`. In kornia when we use/say RGB we mean sRGB and `linear RGB` is the non-gamma-rectified version of sRGB. [ref](https://stackoverflow.com/questions/35952564/convert-rgb-to-srgb)
 -  modified test_lab and test_luv to include edge values [0,1] in the unit tests, since that's where it failed before. I had thought `test_gradcheck` also caught nans in the gradient but doesn't seem so 🤔 Maybe we should do some of the checks on real images vs randomly generated values, but then test costs might be higher.  

LGTM for now

edgarriba reviewed (approved):
LGTM
